app/Global.scala

```diff
@@ -73,65 +76,89 @@ object Global extends GlobalSettings with LazyLogging{
  * Initial set of data to be imported
  * in the sample application.
  */
-object InitialData extends GlobalDBAccess with LazyLogging {
-
-  val mpi = Team("Connectomics department", None, RoleService.roles)
-
-  def insert(conf: Configuration) = {
-    insertDefaultUser(conf)
-    insertTeams()
-    insertTasks()
-    if (conf.getBoolean("datastore.enabled").getOrElse(true)) {
-      insertLocalDataStore(conf)
-    }
-  }
-
-  def insertDefaultUser(conf: Configuration) = {
+object InitialData extends GlobalDBAccess with FoxImplicits with LazyLogging {
+
+  val defaultUserEmail = Play.configuration.getString("application.authentication.defaultUser.email").get
+  val defaultUserPassword = Play.configuration.getString("application.authentication.defaultUser.password").get
+  val rootTeamName = "Connectomics department"
+
+  def insert =
+    for {
+      _ <- insertDefaultUser
+      _ <- insertRootTeam
+      _ <- giveDeafultUserTeam
+      _ <- insertTaskType
+      _ <- insertProject
+      _ <- if (Play.configuration.getBoolean("datastore.enabled").getOrElse(true)) insertLocalDataStore else Fox.successful(())
+    } yield ()
+
+  def insertDefaultUser =  {
     UserService.defaultUser.futureBox.map {
-      case Full(_) =>
+      case Full(_) => Fox.successful(())
       case _ =>
-        val email = conf.getString("application.authentication.defaultUser.email").getOrElse("scmboy@scalableminds.com")
-        val password = conf.getString("application.authentication.defaultUser.password").getOrElse("secret")
+        val email = defaultUserEmail
+        val password = defaultUserPassword
         logger.info("Inserted default user scmboy")
         UserDAO.insert(User(
           email,
           "SCM",
           "Boy",
           true,
           SCrypt.md5(password),
-          List(TeamMembership(mpi.name, Role.Admin)),
+          List(),
           loginInfo = UserService.createLoginInfo(email),
-          passwordInfo = UserService.createPasswordInfo(password))
+          passwordInfo = UserService.createPasswordInfo(password),
+          experiences = Map("sampleExp" -> 10))
         )
-    }
+    }.flatten
   }
 
-  def insertTeams() = {
-    TeamDAO.findOne().futureBox.map {
-      case Full(_) =>
+  def insertRootTeam = {
+    TeamDAO.findOneByName(rootTeamName).futureBox.map {
```

Comment:
where is the root team added to the db?

Reply:
@normanrz three lines below. This just checks if the team exists to not insert it twice